core/posts/views.py

```diff
@@ -1,31 +1,100 @@
-from core.posts.constants import DEFAULT_POST_PAGE_SIZE
-from core.authors.models import Follow
-from core.authors.util import get_author_id
-
-import logging
 
 logger = logging.getLogger(__name__)
 
+COMMENT_NOT_ALLOWED = 'Comment not allowed'
+COMMENT_ADDED = 'Comment Added'
+
+
+def create_comment(request, pk=None):
+    post = Posts.objects.get(pk=pk)
+    if post:
+        data = request.data
+        print(data)
+        comment = data.get('comment', None)
+        author = comment.get('author', None)
+        author_id = author['id'].rstrip('/').split('/')[-1]
+        comment['author'] = author_id
+        serializer = CommentsSerializer(data=comment)
+        if serializer.is_valid():
+            post.comments.create(**serializer.validated_data)
+            return Response({
+                "query": "addComment",
+                "success": True,
+                "message": COMMENT_ADDED
+            }, status=status.HTTP_200_OK)
+        else:
+            return Response({
+                "query": "addComment",
+                "success": False,
+                "message": COMMENT_NOT_ALLOWED,
+                "errors": serializer.errors
+            }, status=status.HTTP_403_FORBIDDEN)
+    else:
+        return Response(status=status.HTTP_404_NOT_FOUND)
+
+
+def list_comments(request, pk=None):
+    size = int(request.query_params.get("size", 5))
+    queryPage = int(request.query_params.get('page', 0))
+    if size < 1 or queryPage < 0 or size > 100:
+        return Response({
+            "success": False,
+            "message": "The query parameters were invalid",
+            "query": "comments"
+        }, 400)
+
+    post = Posts.objects.get(pk=pk)
+    if not can_user_view(request.user, post):
+        return Response({
+            "success": False,
+            "message": "You are not authorized to view this post's comments.",
+            "query": "comments"
+        }, status=401)
+
+    comments = Comments.objects.filter(post=post)
+
+    try:
+        paginator = Paginator(comments, size)
+        page = paginator.page(queryPage + 1)
+        serializer = CommentsSerializer(page, many=True, context={'request': request})
+        comments_to_return = serializer.data
+    except:
+        comments_to_return = []
+
+    data = {
+        "comments": comments_to_return,
+        "query": "comments",
+        "count": len(comments),
+        "size": size
+    }
+    if (len(comments_to_return) > 0):
+        add_page_details_to_response(request, data, page, queryPage)
+
+    return Response(data)
+
 
 class PostsViewSet(viewsets.ModelViewSet):
     queryset = Posts.objects.filter(visibility="PUBLIC").order_by('-published')
     serializer_class = PostsSerializer
 
-    def retrieve(self, request, pk):
-        post = Posts.objects.get(pk=pk)
-        if (not can_user_view(request.user, post)):
+    def retrieve(self, request, *args, **kwargs):
+        post = self.get_object()
```

Comment:
When I last checked this, get_object() only returned Posts that were in the queryset (i.e. if you have a private post you're trying to retrieve then this would not return it)